includes/class-wc-product-variable.php

```diff
@@ -455,11 +455,15 @@ public static function sync( $product, $save = true ) {
 			$data_store->sync_stock_status( $product );
 			self::sync_attributes( $product ); // Legacy update of attributes.
 
-			do_action( 'woocommerce_variable_product_sync', $product->get_id(), $product->get_visible_children(), $save );
+			do_action( 'woocommerce_variable_product_sync_before_save', $product );
 
 			if ( $save ) {
 				$product->save();
 			}
+
+			do_action( 'woocommerce_variable_product_synced', $product );
```

Comment:
The main use case I had in mind was an easy replacement for `'woocommerce_variable_product_sync'`.

I checked for suitable replacements in `WC_Product_Variable::save()`, but there was only the `'woocommerce_before_' . $this->object_type . '_object_save'` hook, which runs before saving.

It looks like further upstream `'woocommerce_new_product'` or `'woocommerce_new_product'` would be suitable though, so I'll remove this and mention those in the deprecated notice for `'woocommerce_variable_product_sync'`.